source/RunScribeDataField.mc

```diff
@@ -203,25 +220,41 @@ class RunScribeDataField extends Ui.DataField {
         
         dc.setColor(fgColor, Gfx.COLOR_TRANSPARENT);
         
+        if (mUpdateLayout != 0) {
+            onLayout(dc);
+        }
+        
         // Update status
-        if (mSensorLeft != null && mSensorRight != null && (mSensorRight.searching == 0 || mSensorLeft.searching == 0)) {
-            var met1x, met1y, met2x, met2y, met3x = 0, met3y = 0, met4x = 0, met4y = 0;
+        if (mSensorLeft != null && mSensorRight != null && mSensorRight.data != null && mSensorLeft.data != null) {

```

Comment:
We must check "searching" property here to be able to display "Searching..." when both channels are scanning. Please, revert back this line. Of course, you can add the check for "data" property as well.

Reply:
data != null check is enough

Reply:
But you never set null to this property, so if the watch loses connectivity there is no way for displaying "Searching...".

Reply:
Searching is handled in the later else branch.

Reply:
For this we use "searching" property as is set to true when we re-open channel.

Reply:
Ok yes, lost connection does not cause "searching" to be displayed, only initial search is visible. If searching can occur later then this should check against "searching".

Reply:
yes, that's correct.  I was talking to Tim yesterday about this and we decided that it's better to show "Searching..." when it occurs.